orm/encoding/ormkv/key_codec.go

```diff
@@ -247,20 +247,19 @@ func (cdc KeyCodec) CheckValidRangeIterationKeys(start, end []protoreflect.Value
 		y := end[i]
 
 		cmp = fieldCdc.Compare(x, y)
-		switch {
-		case cmp > 0:
+		if cmp > 0 {
 			return ormerrors.InvalidRangeIterationKeys.Wrapf(
 				"start must be before end for field %s",
 				cdc.fieldDescriptors[i].FullName(),
 			)
-		case !fieldCdc.IsOrdered() && cmp != 0:
+		} else if !fieldCdc.IsOrdered() && cmp != 0 {
 			descriptor := cdc.fieldDescriptors[i]
 			return ormerrors.InvalidRangeIterationKeys.Wrapf(
 				"field %s of kind %s doesn't support ordered range iteration",
 				descriptor.FullName(),
 				descriptor.Kind(),
 			)
-		case cmp < 0:
+		} else if cmp < 0 {
```

Comment:
This bug was introduced in #15813

Reply:
I am curious why is this a bug? This behaves the same, right?
Golangci-lint complains as well then

Reply:
This changes the target of the break statement and makes the loop incorrect. The linter is buggy

Reply:
I see, the diff does not show the loop.